mods/fire/init.lua

```diff
@@ -78,15 +78,20 @@ minetest.register_tool("fire:flint_and_steel", {
 		local pt = pointed_thing
 
 		if pt.type == "node" and minetest.get_node(pt.above).name == "air" then
-			if not minetest.is_protected(pt.above, player_name) then
-				minetest.set_node(pt.above, {name="fire:basic_flame"})
-			else
-				minetest.chat_send_player(player_name, "This area is protected")
+			itemstack:add_wear(1000)
+			local node_under = minetest.get_node(pt.under).name
+
+			if minetest.get_node_group(node_under, "flammable") >= 1 and
+					minetest.registered_nodes[node_under].drawtype == "normal" then
```

Comment:
This would mean that stairs, slabs, fences and many other nodes made from wood will not burn. Wood nodes should burn no matter what the drawtype is.